src/dags/grex.py

```diff
@@ -0,0 +1,128 @@
+from environs import Env
+import pandas as pd
+from sqlalchemy import create_engine
+from sqlalchemy.types import Integer, Text, Float, Date
+from geoalchemy2 import Geometry, WKTElement
+from shapely.geometry.multipolygon import MultiPolygon
+from shapely.geometry.polygon import Polygon
+from shapely import wkt
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.postgres_operator import PostgresOperator
+from postgres_check_operator import PostgresCheckOperator, PostgresValueCheckOperator
+from airflow.operators.python_operator import PythonOperator
+
+from common import default_args
+from swift_operator import SwiftOperator
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+
+from common.sql import (
+    SQL_CHECK_COUNT,
+    SQL_CHECK_GEO,
+)
+
+
+def wkt_loads_wrapped(data):
+    p = wkt.loads(data)
+    if isinstance(p, Polygon):
+        p = MultiPolygon([p])
+    elif isinstance(p, MultiPolygon):
+        pass
+    else:
+        p = None
+    if p:
+        p = WKTElement(p.wkt, srid=4326)
+    return p
+
+
+def load_grex(input_csv, table_name):
+    env = Env()
+    user = env("POSTGRES_USER")
+    password = env("POSTGRES_PASSWORD")
+    host = env("POSTGRES_HOST")
+    port = env("POSTGRES_PORT")
+    db = env("POSTGRES_DB")
+
+    db_engine = create_engine(f"postgresql://{user}:{password}@{host}:{port}/{db}")
+    df = pd.read_csv(
+        input_csv,
+        delimiter=";",
+        decimal=",",
+        encoding="latin_1",
+        parse_dates=["STARTDATUM"],
+        dayfirst=True,
+        index_col="PLANNR",
+    )
+    df.rename(columns=lambda x: x.lower(), inplace=True)
+    df.index.name = "id"
+    df["geometrie"] = df["geometrie"].apply(wkt_loads_wrapped)
+
+    grex_rapportage_dtype = {
+        "id": Integer(),
+        "plannaam": Text(),
+        "planbeheercode": Text(),
+        "startdatum": Date(),
+        "planstatus": Text(),
+        "oppervlakte": Float(),
+        "geometrie": Geometry(geometry_type="MULTIPOLYGON", srid=4326),
+    }
+
+    df.to_sql(table_name, db_engine, if_exists="replace", dtype=grex_rapportage_dtype)
+    with db_engine.connect() as connection:
+        connection.execute("ALTER TABLE grex_new ADD PRIMARY KEY (id)")
+
+
+dag_id = "grex"
+dag_config = Variable.get(dag_id, deserialize_json=True)
+
+table_name = dag_id
+table_name_new = f"{dag_id}_new"
+SQL_TABLE_RENAME = f"""
```

Comment:
Deze zou evt. ook nog kunnen met de jinja2 templating ipv. f-strings. Maar, in dit geval maakt dat verder niet zoveel uit.